hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/api/OMDBInsightEndpoint.java

```diff
@@ -0,0 +1,454 @@
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ * <p>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.hadoop.ozone.recon.api;
+
+import org.apache.commons.lang3.StringUtils;
+import org.apache.hadoop.hdds.protocol.proto.HddsProtos;
+import org.apache.hadoop.hdds.scm.container.ContainerInfo;
+import org.apache.hadoop.hdds.scm.server.OzoneStorageContainerManager;
+import org.apache.hadoop.hdds.utils.db.Table;
+import org.apache.hadoop.hdds.utils.db.TableIterator;
+import org.apache.hadoop.ozone.om.helpers.BucketLayout;
+import org.apache.hadoop.ozone.om.helpers.OmKeyInfo;
+import org.apache.hadoop.ozone.om.helpers.RepeatedOmKeyInfo;
+import org.apache.hadoop.ozone.recon.api.types.ContainerMetadata;
+import org.apache.hadoop.ozone.recon.api.types.KeyEntityInfo;
+import org.apache.hadoop.ozone.recon.api.types.KeyInsightInfoResp;
+import org.apache.hadoop.ozone.recon.api.types.KeysResponse;
+import org.apache.hadoop.ozone.recon.recovery.ReconOMMetadataManager;
+import org.apache.hadoop.ozone.recon.scm.ReconContainerManager;
+import org.apache.hadoop.ozone.recon.spi.ReconContainerMetadataManager;
+
+import javax.inject.Inject;
+import javax.ws.rs.DefaultValue;
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.QueryParam;
+import javax.ws.rs.WebApplicationException;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import static org.apache.hadoop.ozone.recon.ReconConstants.DEFAULT_FETCH_COUNT;
+import static org.apache.hadoop.ozone.recon.ReconConstants.RECON_QUERY_LIMIT;
+import static org.apache.hadoop.ozone.recon.ReconConstants.RECON_QUERY_PREVKEY;
+
+/**
+ * Endpoint to get following key level info under OM DB Insight page of Recon.
+ * 1. Number of open keys for Legacy/OBS buckets.
+ * 2. Number of open files for FSO buckets.
+ * 3. Amount of data mapped to open keys and open files.
+ * 4. Number of pending delete keys in legacy/OBS buckets and pending
+ * delete files in FSO buckets.
+ * 5. Amount of data mapped to pending delete keys in legacy/OBS buckets and
+ * pending delete files in FSO buckets.
+ */
+@Path("/omdbinsight")
```

Comment:
We may not need  /omdbinsight in path

Reply:
> We may not need /omdbinsight in path

@krishnaasawa1  this is the general convention that what this whole endpoint is about and what the APIs contains regarding, we follow this convention in other endpoints also like "MetricsProxyEndpoint",  "NSSummaryEndpoint", "ClusterStateEndpoint" etc.

Reply:
Those belong to /metrics , /namespace and /clusterstate APIs , so here new APIs are for Containers and Keys. Based on https://ozone.apache.org/docs/current/interface/reconapi.html it looks more appropriate to keep them in Containers and Keys category respectively as information is regarding that instead of creating appending new categories  omdbinsight , scminsights , insights etc in path APIs. Those APIs can map in UI accordingly to whatever insights we want to display to user. e.g namespace apis are called from insights UI page.

Reply:
> Those belong to /metrics , /namespace and /clusterstate APIs , so here new APIs are for Containers and Keys. Based on https://ozone.apache.org/docs/current/interface/reconapi.html it looks more appropriate to keep them in Containers and Keys category respectively as information is regarding that instead of creating appending new categories omdbinsight , scminsights , insights etc in path APIs. Those APIs can map in UI accordingly to whatever insights we want to display to user. e.g namespace apis are called from insights UI page.

This point can be debatable, I have removed the "omdbinsight" from top path of API,  however based on your earlier comment, I changed the API path to "/containers/keys/mismatch" for the API which retrieves set of keys/files/dirs which are mapped to
   * containers in DELETED state in SCM. so now based on above comment, we should not keep the API path as "/containers/keys/mismatch", because this API doesn't retrieve containers, so I have changed the API path by adding top level as "/keys" and rename the path like "/containers/mismatch".

Reply:
As discussed, I have moved "/containers/mismatch/keys" to ContainerEndPoint, pls have a re-look.